src/controller/CHIPDeviceController.cpp

```diff
@@ -2014,32 +1989,36 @@ void DeviceCommissioner::OnDone(app::ReadClient *)
     // Try to parse as much as we can here before returning, even if this is an error.
     return_err = err == CHIP_NO_ERROR ? return_err : err;
 
-    // Set the network cluster endpoints first so we can match up the connection times.
+    // Set the network cluster endpoints first so we can match up the connection
+    // times.  Note that here we don't know what endpoints the network
+    // commissioning clusters might be on.
     err = mAttributeCache->ForEachAttribute(
         app::Clusters::NetworkCommissioning::Id, [this, &info](const app::ConcreteAttributePath & path) {
-            if (path.mAttributeId != app::Clusters::NetworkCommissioning::Attributes::FeatureMap::Id)
+            using namespace chip::app::Clusters::NetworkCommissioning;
+            using namespace chip::app::Clusters::NetworkCommissioning::Attributes;
+            if (path.mAttributeId != FeatureMap::Id)
             {
                 return CHIP_NO_ERROR;
             }
             TLV::TLVReader reader;
             if (this->mAttributeCache->Get(path, reader) == CHIP_NO_ERROR)
             {
-                BitFlags<app::Clusters::NetworkCommissioning::Feature> features;
+                BitFlags<Feature> features;
```

Comment:
How about just `using namespace chip::app::Clusters` and explicitly adding `NetworkCommissioning` in front of every Feature ? The generic `Feature` looks the same for all clusters and it is more painful to grep and even human-validate for. At least `kWiFiNetworkInterface` and such is unique, however I would still prefer to see `NetworkCommissioning::Feature` or `NetworkCommissioning::FeatureMap` for clarity.

Reply:
I can do that.